### DISCUSSION TITLE
How to stop if stalled when execute run_time or run_angle?

### DISCUSSION THREAD

**Discussion by sgatu** [pybricks/support#1185] [Q&A] [answered]
Hi,

Is there any why to stop if stalled when run_time or run_angle is executed?  And if so, how feasible is to be able to change between the two behaviors? 

From my test it looks like stall_tolerances have no effect on those 2 methods, and neither for run_until_stalled, which will stop if stalled independently of stall_tolerances configuration.

**Comment by laurensvalk** [comment, marked as answer]
In Pybricks, being `stalled` is only  a flag that gets set; it doesn't abort any ongoing comands.

`run_until_stalled` is a convenience function that does `run(...)` and then waits until it stalls. You could make variants for each of the other commands as well. For example:

```python

# Start rotating, but don't wait for completion
my_motor.run_angle(500, 360, wait=False)

# While not done, check the stall state.
while not my_motor.done():
    wait(5)
    # Stop on stall. This also sets it to done so the loop ends.
    if my_motor.stalled():
        my_motor.stop()

```

**Reply by sgatu** [reply]
This helped me with the development, the code is a bit more complex due to using multitask in beta, but I got the functionality I wanted. Thank you very much.
